src/runtime_src/core/edge/drm/zocl/sched_exec.c

```diff
@@ -801,6 +798,27 @@ configure(struct sched_cmd *cmd)
 		exec->configured = 1;
 	} else {
 		SCHED_DEBUG("++ configuring PS ERT mode\n");
+
+		/*
+		 * Block comment for the big picture of ert with static or
+		 * dynamic xclbins.
+		 *   - versal ERT defer dynamic region xclbin download, so we
+		 *     call init_cu_soft in configure and defer the
+		 *     init_cu_hard till pdi is downloaded via softkernel.
+		 *   - other ERT will load xclbin first then call configure, so
+		 *     that we call init_cu_full in configure.
+		 */
+		if (zdev->ert->ops->static_xclbin()) {
+			exec->configured = 1;
+		} else {
+			/*
+			 * dynamic xclbin like versal, the procedure is:
+			 * loalxclbin->scheduler:init->configure->softkernel_config.
```

Comment:
I think the better wording would be replace softkernel_config to pdi download because this is not really soft kerenl. We just happen to use configure soft kernel (opcode) to download PDI. So download PDI would be more straight forword.